phoenix-core-client/src/main/java/org/apache/phoenix/schema/RowKeyValueAccessor.java

```diff
@@ -37,14 +47,58 @@
  * 
  * @since 0.1
  */
-public class RowKeyValueAccessor implements Writable   {
+public class RowKeyValueAccessor implements Writable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RowKeyValueAccessor.class);
+
     /**
      * Constructor solely for use during deserialization. Should not
      * otherwise be used.
      */
     public RowKeyValueAccessor() {
     }
-    
+
+    /**
+     * The class used to keep list of booleans representing whether the data type of the given
+     * field is of type VARBINARY_ENCODED. For example, let's say we have variable length
+     * data types in the order (VARCHAR, VARBINARY_ENCODED, VARCHAR). The corresponding object
+     * of this class will have binaryEncodedDataTypes list with values (false, true, false).
+     * The array of this class is serialized and deserialized together with other fields of
+     * RowKeyValueAccessor. The value of the list determines which separator bytes need to be
+     * used while iterating through the rowkey bytes.
+     */
+    static class BinaryEncodedTypesLists {
```

Comment:
BinaryEncodedTypesLists does not tell what this is about.  It gets confusing when you declare BinaryEncodedTypesLists[] binaryEncodedTypesLists later.  I think ListOfBinaryEncodedTypeFlags or something similar would be more readable. Also did you consider using BitSet? If you use it, you do not need these two classes.